src/main/java/space/obminyashka/items_exchange/service/UserService.java

```diff
@@ -62,23 +55,24 @@ public boolean existsByUsernameOrEmailAndPassword(String usernameOrEmail, String
     }
 
     public boolean registerNewUser(UserRegistrationDto userRegistrationDto, Role role) {
-        User registeredUser = UserMapper.userRegistrationDtoToUser(userRegistrationDto, bCryptPasswordEncoder, role);
+        User registeredUser = userRegistrationDtoToUser(userRegistrationDto, bCryptPasswordEncoder, role);
         return userRepository.save(registeredUser).getId() != 0;
     }
 
-    public UserDto update(UserDto newUserDto, User user) throws IllegalOperationException {
-        User updatedUser = convertDto(newUserDto);
-        var newChildren = extractAll(updatedUser.getChildren(), child -> child.getId() == 0, ArrayList::new);
-        var newPhones = extractAll(updatedUser.getPhones(), phone -> phone.getId() == 0, HashSet::new);
+    public String update(UserUpdateDto newUserUpdateDto, User user) {
+        user.setFirstName(newUserUpdateDto.getFirstName());
+        user.setLastName(newUserUpdateDto.getLastName());
 
-        checkIsAllowedToAddNewChildrenOrPhones(user, !newChildren.isEmpty(), !newPhones.isEmpty());
-        checkReadOnlyFieldsUpdate(updatedUser, user);
-
-        BeanUtil.copyProperties(updatedUser, user, "email", "firstName", "lastName", "avatarImage");
-        user.setUpdated(LocalDateTime.now());
-        addNewChildren(user, newChildren);
-        addNewPhones(user, newPhones);
-        return mapUserToDto(userRepository.saveAndFlush(user));
+        final var phonesToUpdate = convertPhone(newUserUpdateDto.getPhones());
+        final var userPhones = user.getPhones();
+        boolean isEqualsPhones = userPhones.equals(phonesToUpdate);
+        if (!isEqualsPhones) {
+            userPhones.removeIf(Predicate.not((phonesToUpdate::contains)));
+            userPhones.addAll(phonesToUpdate);
+            userPhones.forEach((phone -> phone.setUser(user)));
+        }
+        convertTo(userRepository.saveAndFlush(user), UserUpdateDto.class);
```

Comment:
You don't have to convert User into DTO again because you don't need that DTO. Just save the user.